code-scanning/codegurureviewer.yml

```diff
@@ -0,0 +1,55 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support
+# documentation.
+#
+# For more information on Amazon CodeGuru Reviewer in general, see
+# https://github.com/aws-actions/codeguru-reviewer
+
+name: CodeGuru Reviewer
+
+on:
+  push:
+    branches: [ $default-branch, $protected-branches ]
+  pull_request:
+    branches: [ $default-branch ]
+  schedule:
+    - cron:  $cron-weekly
+  workflow_dispatch:
+    inputs:
+      logLevel:
+        description: 'Log level'
+        required: true
+        default: 'warning'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+  steps:
+    # Step 1: Checkout the repository and provide your AWS credentials
+    - name: Checkout repository
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@0d9a5be0dceea74e09396820e1e522ba4a110d2f
+      with:
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        aws-region: us-west-2  # Region to access CodeGuru
+
+    # Step 2: Add CodeGuru Reviewer Action
+    - name: CodeGuru Reviewer
+      uses: aws-actions/codeguru-reviewer@e96af14e3ba9bac3cca62a524bd5a8345038c684
+      with:
+        build_path: target # build artifact(s) directory
+        s3_bucket: codeguru-reviewer-my-bucket  # S3 Bucket with "codeguru-reviewer-*" prefix
+
+    # Step 3: Upload results into GitHub
+    - name: Upload review result
+      if: ${{ github.event_name != 'push' }}
```

Comment:
```suggestion
```

Is there a reason why this limitation is included? Code scanning relies on results being uploaded for `pull_request` events in order to provide results on PRs.

Reply:
This was added because for `push` events, our code scanner only runs a scan on the changed lines of code. Thus, some recommendations that were populated in the security tab from a scheduled scan would get closed when a push scan was run (as the new SARIF file would only contain recommendations on the changed lines). 

We do still upload the results for `pull_request` events, this is just excluding `push` events. Example seen here: https://github.com/aschembo-dev/testcicd/pull/158

Reply:
Sorry, yes, it was late and I got that backwards. 😄

We do rely on both `push` and `pull_request` analyses to provide results on PRs though, letting us determine which alerts have been introduced in that PR (relative to the baseline analysis).

Is your tool already accounting for that in some way?

Reply:
The issue is that a `push` event scan will mark existing recommendations (in the security tab) as closed even if they may not have been fixed due to our `push` scan having recommendations only on the changed lines of code. 

I can understand that for new alerts to be added in a PR, there needs to be a `push` event then, but we are already providing these recommendations in the comment of the PR, and the `push` event could close existing ones. 

This is the reason we disabled `push` events from uploading SARIF file.